pkg/querier/handler.go

```diff
@@ -0,0 +1,102 @@
+package querier
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/queryrange"
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+)
+
+type Handler struct {
+	api *QuerierAPI
+}
+
+func NewQuerierHandler(api *QuerierAPI) *Handler {
+	return &Handler{
+		api: api,
+	}
+}
+
+func (h *Handler) Do(ctx context.Context, req queryrangebase.Request) (queryrangebase.Response, error) {
+	switch concrete := req.(type) {
+	case *queryrange.LokiRequest:
+		res, err := h.api.RangeQueryHandler(ctx, concrete)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err := queryrange.ParamsFromRequest(req)
+		if err != nil {
+			return nil, err
+		}
+
+		return queryrange.ResultToResponse(res, params)
+	case *queryrange.LokiInstantRequest:
+		res, err := h.api.InstantQueryHandler(ctx, concrete)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err := queryrange.ParamsFromRequest(req)
+		if err != nil {
+			return nil, err
+		}
+
+		return queryrange.ResultToResponse(res, params)
+	case *queryrange.LokiSeriesRequest:
+		request := &logproto.SeriesRequest{}
+		result, statResult, err := h.api.SeriesHandler(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		return &queryrange.LokiSeriesResponse{
+			Status:     "success",
+			Version:    uint32(loghttp.VersionV1),
+			Data:       result.Series,
+			Statistics: statResult,
+		}, nil
+	case *queryrange.LokiLabelNamesRequest:
+		// TODO: LokiLabelNamesRequest should probably be logproto.LabelRequest
```

Comment:
can remove this TODO now

Reply:
Will be addressed in https://github.com/grafana/loki/pull/10956